gateway-topology-simple/src/main/java/org/apache/knox/gateway/topology/simple/SimpleDescriptorHandler.java

```diff
@@ -571,13 +579,49 @@ private static File resolveProviderConfigurationReference(final String reference
             if (topologyFilename == null) {
                 topologyFilename = desc.getCluster();
             }
+
             topologyDescriptor = new File(destDirectory, topologyFilename + ".xml");
 
-            try (OutputStream outputStream = Files.newOutputStream(topologyDescriptor.toPath());
-                 OutputStreamWriter outputStreamWriter = new OutputStreamWriter(outputStream, StandardCharsets.UTF_8);
-                 BufferedWriter fw = new BufferedWriter(outputStreamWriter)) {
-              fw.write(sw.toString());
-              fw.flush();
+            // KNOX-2302
+            boolean persistGenerated = false;
+            // Determine if the topology already exists.
+            if (topologyDescriptor.exists()) {
```

Comment:
I'd move this new logic to its own private method with a meaningful name (e.g. `shouldRedeployTopology`) to make the code cleaner.

Reply:
+1